Cargo.toml

```diff
@@ -7,7 +7,6 @@ edition = "2021"
 
 [dependencies]
 ratatui = "0.27"
-rayon = "1.10.0"
```

Comment:
This is to test. At least for me, every usage of rayon was faster removed.

Reply:
I tested it again many times without rayon and I got a pretty consistant 12% reduction in performance when removing rayon without the other optimization. 
![reduction in performance without rayon](https://github.com/user-attachments/assets/813af69f-33b8-4bf8-ae6f-0b4984e89e30)

I think that you should try it with a baseline and maybe some more measurement time if you have a lot of outliers

```shell
cargo bench --bench bench_main -- --verbose --measurement-time 15 --save-baseline with-rayon
# remove rayon here
cargo bench --bench bench_main -- --verbose --measurement-time 15 --baseline with-rayon
```

It may WSL like you said but I think it's better to keep it since it requires more testing on different machines / it should in theory be faster since it is changing values that can be done in parallel / it already works faster with rayon on my machine.